staging/src/slime.io/slime/modules/lazyload/controllers/update_resources.go

```diff
@@ -0,0 +1,185 @@
+	Limiter *config.Limiter    `protobuf:"bytes,2,opt,name=limiter,proto3" json:"limiter,omitempty"`
+	Global  *config.Global     `protobuf:"bytes,3,opt,name=global,proto3" json:"global,omitempty"`
+	Fence   *config.Fence      `protobuf:"bytes,4,opt,name=fence,proto3" json:"fence,omitempty"`
+	Metric  *config.Metric     `protobuf:"bytes,6,opt,name=metric,proto3" json:"metric,omitempty"`
+	Name    string             `protobuf:"bytes,5,opt,name=name,proto3" json:"name,omitempty"`
+	Enable  bool               `protobuf:"varint,7,opt,name=enable,proto3" json:"enable,omitempty"`
+	General *v1alpha1.Fence    `protobuf:"bytes,8,opt,name=general,proto3" json:"general,omitempty"`
+	Bundle  *config.Bundle     `protobuf:"bytes,9,opt,name=bundle,proto3" json:"bundle,omitempty"`
+	Mode    config.Config_Mode `protobuf:"varint,10,opt,name=mode,proto3,enum=slime.config.v1alpha1.Config_Mode" json:"mode,omitempty"`
+	// like bundle item kind, necessary if not bundle
+	Kind string `protobuf:"bytes,11,opt,name=kind,proto3" json:"kind,omitempty"`
+}
+
+func updateResources(wormholePort []string, env bootstrap.Environment) {
+
+	log := log.WithField("function", "updateResources")
+	cliSet := env.K8SClient
+	dynCli := env.DynamicClient
+
+	// get slimeboot cr name
+	slimeBootNs := os.Getenv("WATCH_NAMESPACE")
+	deployName := strings.Split(os.Getenv("POD_NAME"), "-")[0]
+	deploy, err := cliSet.AppsV1().Deployments(slimeBootNs).Get(deployName, metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("get lazyload deployment [%s/%s] error: %+v", slimeBootNs, deployName, err)
+		return
+	}
+	slimeBootName := deploy.OwnerReferences[0].Name
+
+	// Unstructured
+	utd, err := dynCli.Resource(slimeBootGvr).Namespace(slimeBootNs).Get(slimeBootName, metav1.GetOptions{}, "")
+	if err != nil {
+		log.Errorf("get slimeboot [%s/%s] error: %+v", slimeBootNs, slimeBootName, err)
+		return
+	}
+	// Unstructured -> SlimeBoot
+	var slimeBoot SlimeBoot
+	if err = runtime.DefaultUnstructuredConverter.FromUnstructured(utd.UnstructuredContent(), &slimeBoot); err != nil {
+		log.Errorf("convert slimeboot %s/%s to structured error: %v", slimeBootNs, slimeBootName, err)
+		return
+	}
+
+	sort.Strings(wormholePort)
+	log.Debugf("sorted wormholePort: %v", wormholePort)
+
+	for _, module := range slimeBoot.Spec.Modules {
+		if module.Kind != "lazyload" {
+			continue
+		}
+		// update wormholePort
+		module.General.WormholePort = wormholePort
+		// add default value
+		module.addDefaultValue()
+	}
+
+	// add default value
+	slimeBoot.addDefaultValue()
+
+	// update slimeBoot
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&slimeBoot)
+	if err != nil {
+		log.Errorf("convert slimeboot %s/%s to unstructured error: %+v", slimeBootNs, slimeBootName, err)
+		return
+	}
+	utd.SetUnstructuredContent(obj)
+	utd, err = dynCli.Resource(slimeBootGvr).Namespace(slimeBootNs).Update(utd, metav1.UpdateOptions{})
```

Comment:
this code seems to do not respect the resourceVersion mechanism and may cause it can only run in single-pod mode.